shm/Game.cpp

```diff
@@ -0,0 +1,179 @@
+#include "Game.hpp"
+#include "Coordinates.hpp"
+#include <cstdio>
+#include <iomanip>
+#include <iostream>
+#include "Map.hpp"
+
+Game::Game(size_t money, size_t days, size_t finalGoal)
+    : money_(money), days_(days), finalGoal_(finalGoal), currentDay_(1) {
+    std::unique_ptr<Time> time(new Time());
+    time_ = std::move(time);
+    std::unique_ptr<Map> map(new Map(time_.get()));
+    map_ = std::move(map);
+    std::unique_ptr<Ship> ship(new Ship(1000, 50, 15, "HMS Terror", 1));
+    std::unique_ptr<Player> player(new Player(ship, money_));
+    player_ = std::move(player);
+    player_->getShip()->setDelegate(player_.get());
+    time_->AddObserver(player_->getShip());
+}
+
+void Game::startGame() {
+    while (true) {
+        checkWinCondition();
+        checkLooseCodition();
+        printStats();
+        printMenu();
+        makeAction(readCharacter());
+    }
+};
+
+void Game::printStats() {
+    std::cout << std::setw(99) << std::setfill('-') << "\n";
+    std::cout << "|" << std::setfill(' ') << std::setw(97) << "|" << "\n";
+    std::cout << "|" << std::setfill(' ') << std::setw(15) << "YOUR MONEY: " << std::setw(8) << std::setfill('0') << money_;
+    std::cout << std::setfill(' ') << std::setw(30) << "YOUR GOAL: " << std::setw(8) << std::setfill('0') << finalGoal_;
+    std::cout << std::setfill(' ') << std::setw(30) << "DAYS LEFT: " << std::setw(3) << std::setfill('0') << days_ - currentDay_ << std::setfill(' ') << std::setw(3) << "|" << "\n";
+    std::cout << "|" << std::setfill(' ') << std::setw(97) << "|" << "\n";
+    std::cout << std::setw(99) << std::setfill('-') << "\n" << "\n";
+    std::cout << (*map_) << "\n";
+    auto island = map_->getCurrentIsland();
+    std::cout << "Island coordinates" << island->getPosition() << "\n";
+    auto store = map_->getCurrentIsland()->getStore();
+    std::cout << (*store);
+
+    player_->printCargo();
+}
+
+void Game::printMenu() {
+    std::cout << std::setw(99) << std::setfill('-') << "\n";
+    std::cout << "|" << std::setfill(' ') << std::setw(56) << "ACTION  MENU" << std::setfill(' ') << std::setw(41) << "|"
+              << "\n";
+    std::cout << std::setw(99) << std::setfill('-') << "\n";
+    std::cout << "1. Buy"
+              << "\n";
+    std::cout << "2. Sell"
+              << "\n";
+    std::cout << "3. Travel"
+              << "\n"
+              << "\n";
+    std::cout << "9. Options"
+              << "\n";
+    std::cout << "10. Exit Game"
+              << "\n";
+    std::cout << std::setw(99) << std::setfill('-') << "\n";
+
+    return;
+}
+
+void Game::makeAction(const Action& action) {
+    switch (action) {
+    case Action::buy:
+        buy();
+        break;
+    case Action::sell:
+        sell();
+        break;
+    case Action::travel:
+        travel();
+        break;
+    case Action::options:
+        //printOptions();
+        break;
+    case Action::exit:
+        exit(0);
```

Comment:
Hmmm better mark some `exit_` flag and check in while loop above. Maybe in future you want to allow player save game or sth and here you imiediatelly exit program.